roc/version.go

```diff
@@ -72,3 +79,43 @@ func parseVersion(s string) SemanticVersion {
 
 	return v
 }
+
+// Validate version compatibility of roc bindings and native library.
+// This function must be called at all library entry point atleast once.
+// Entry points refer to exported non-method functions of this package
+// This includes OpenContext, OpenSender, OpenReceiver, ParseEndpoint,
+// SetLogger, SetLoggerFunc and SetLogLevel
+func (version Versions) Validate() error {
+	bindingsVersion := version.Bindings
+	nativeVersion := version.Native
+
+	errMsg := fmt.Sprintf(
+		"Detected incompatibility between roc bindings ( %s ) and native library ( %s ): ",
+		bindingsVersion.String(), nativeVersion.String(),
+	)
+
+	if nativeVersion.Major != bindingsVersion.Major {
+		return fmt.Errorf(errMsg + "Major versions are different")
+	}
+
+	if nativeVersion.Minor > bindingsVersion.Minor {
+		return fmt.Errorf(errMsg + "Minor version of binding is less than native library")
+	}
+
+	return nil
+}
+
+var versionCheckOnce int32
+
+// Runs check for version compatibility
+func versionCheck() {
+	if atomic.CompareAndSwapInt32(&versionCheckOnce, 1, 0) {
```

Comment:
It should be `CompareAndSwapInt32(&versionCheckOnce, 0, 1)` because initial value of versionCheckOnce is `0`.